kube/apis/v1alpha1/types_connector.go

```diff
@@ -0,0 +1,114 @@
+}
+
+// +kubebuilder:object:root=true
+type ConnectorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []ConnectorList `json:"items"`
+}
+
+// ConnectorSpec defines the desired state of a ConnectorSpec.
+type ConnectorSpec struct {
+	// SubnetRouter configures a Tailscale subnet router to be deployed in
+	// the cluster. If this is unset no subnet router will be deployed.
+	// https://tailscale.com/kb/1019/subnets/
+	SubnetRouter *SubnetRouter `json:"subnetRouter,omitempty"`
+}
+
+// SubnetRouter describes a subnet router.
+// If unset none will be deployed.
+type SubnetRouter struct {
+	// Routes are the in-cluster CIDRs that the subnet router should make
+	// available. Routes must be within the cluster Pod CIDR range (full
+	// range or a subset)
+	Routes []string `json:"routes"`
+	// Tag that the Tailscale node will be tagged with. If you want the
+	// subnet router to be autoapproved, you can configure Tailscale ACLs to
+	// autoapprove the routes defined in CIDRs for this tag.
+	// See https://tailscale.com/kb/1018/acls/#auto-approvers-for-routes-and-exit-nodes
+	// Defaults to tag-k8s.
+	// If you specify a custom tag here, you must also make tag:k8s-operator owner of the custom tag.
+	// See  https://tailscale.com/kb/1236/kubernetes-operator/#setting-up-the-kubernetes-operator
+	// +optional
+	Tag string `json:"tag,omitempty"`
+}
+
+// ConnectorStatus defines the observed state of the Connector.
+type ConnectorStatus struct {
+
+	// List of status conditions to indicate the status of the Connector.
+	// Known condition types are `SubnetRouterReady`.
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []ConnectorCondition `json:"conditions,omitempty"`
+}
+
+// ConnectorCondition contains condition information for a Connector.
+type ConnectorCondition struct {
+	// Type of the condition, known values are (`SubnetRouterReady`).
+	Type ConnectorConditionType `json:"type"`
+
+	// Status of the condition, one of ('True', 'False', 'Unknown').
+	Status metav1.ConditionStatus `json:"status"`
+
+	// LastTransitionTime is the timestamp corresponding to the last status
+	// change of this condition.
+	// +optional
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// Reason is a brief machine readable explanation for the condition's last
+	// transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// Message is a human readable description of the details of the last
+	// transition, complementing reason.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// If set, this represents the .metadata.generation that the condition was
+	// set based upon.
+	// For instance, if .metadata.generation is currently 12, but the
+	// .status.condition[x].observedGeneration is 9, the condition is out of date
+	// with respect to the current state of the Connector.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```

Comment:
Any external observer can use this to verify that the Ready status refers to the latest spec version